python/lib/client/dmod/client/request_clients.py

```diff
@@ -264,9 +266,27 @@ class DatasetExternalClient(DatasetClient,
     """
 
     # In particular needs - endpoint_uri: str, ssl_directory: Path
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, cache_session: bool = True, **kwargs):
         super().__init__(*args, **kwargs)
-        self._cached_session_file = Path.home().joinpath('.dmod_client_session')
+        self._cached_session_file: Optional[Path] = (
+            Path.home().joinpath(".dmod_client_session") if cache_session else None
+        )
```

Comment:
This wasn't the best `__init__` implementation.  It nullifies logic in the superclass for receiving a session file path, favoring a default value over a user-supplied one.  I don't remember if there was some reason other than laziness at this level.

Regardless, given there is logic in the _ExternalClient_ for __cached_session_file_, should we perhaps do this there?  Or, to phrase in a way that may be more scope-appropriate, why exactly does this type need to offer the option to not cache a session to a file.

Reply:
> Regardless, given there is logic in the _ExternalClient_ for __cached_session_file_, should we perhaps do this there? Or, to phrase in a way that may be more scope-appropriate, why exactly does this type need to offer the option to not cache a session to a file.

My intention was to use this client to create the rest proxy data service. In the context of that use case, an on disk session cache kind of gets in the way when all I really need to do is pass along an existing session's information.

My original plan was to remove the responsibility of caching and session management from the `_ExternalClient_` and introduce `AuthClient` and `CachedAuthClient` implementations. These objects would handle session logic like, for example, adding auth data to requests and in the future refreshing sessions. An external client could either have a `AuthClient` injected or create a default `AuthClient` object if one is not injected. However, this refactor was larger than I had anticipated and I am not confident that I could make theses changes without breaking existing behavior or introducing subtle bugs given the timeframe. So, I opted for the simplest solution.